src/main/scala/is/hail/expr/ir/Pretty.scala

```diff
@@ -190,19 +182,11 @@ object Pretty {
             case In(i, typ) => s"${ typ.parsableString() } $i"
             case Die(message, typ) => typ.parsableString() + " " + prettyStringLiteral(message)
             case Uniroot(name, _, _, _) => prettyIdentifier(name)
-            case MatrixRead(typ, _, _, dropCols, dropRows, MatrixNativeReader(path, spec)) =>
-              prettyStringLiteral(path) + " " +
-                prettyBooleanLiteral(dropCols) + " " +
-                prettyBooleanLiteral(dropRows) + " " +
-                (if (typ == spec.matrix_type)
-                  "None"
-                else
-                  typ.parsableString())
-            case MatrixRead(typ, _, _, dropCols, dropRows, MatrixRangeReader(nRows, nCols, nPartitions)) =>
-              s"$nRows $nCols" + " " +
-                nPartitions.map(_.toString).getOrElse("None") + " " +
-                prettyBooleanLiteral(dropCols) + " " +
-                prettyBooleanLiteral(dropRows)
+            case MatrixRead(typ, dropCols, dropRows, reader) =>
+              typ.getOrElse(reader.fullType).parsableString() + " " +
```

Comment:
Actually, shouldn't this output None for the type if it is None?

Reply:
ahh yes

Reply:
will fix